### PR TITLE
Reordered composer requirements

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targeting this branch, because this applies for this new release.
Also this could be done for 1.x so maybe we need to move this PR to 1.x branch first.

## Changelog

```markdown
### Fixed
- Add PHP and Psr/log as required dependencies
```
## Subject

At first I wanted to update PHP from 7.0 to 7.1, then I saw the other things:

* Add Sonata contributors as authors
* Move PHP to required section
* Move psr/log to required section (not sure about this one)
* Sort composer packages

### CODE REVIEW

**Comment by greg0ire** [comment]
You should have done one commit per item in your list, using `git add -p` Also "reored" => "reordered"

**Comment by greg0ire** [comment]
`psr/log` was introduced in 7e09191a102143164294454abe1e92c65363a9da, and no reason was given for doing that. I think it is safe to move it to the `require` section without giving it too much thought though, because it's a very common and lightweight dependency to have.

**Comment by greg0ire** [comment]
> Remove autoload-dev (was removed on AdminBundle a long ago)

Please explain why you did that in the commit message. I think it is because phpunit does not need autoload to work.

File: composer.json [hunk 1]
```diff
@@ -32,13 +38,13 @@
     "autoload": {
         "psr-4": { "Sonata\\Cache\\": "src/" }
     },
-    "autoload-dev": {
```
Comment (core23) [inline composer.json]: Why did you remove this?

File: composer.json [hunk 2]
```diff
@@ -10,17 +10,23 @@
...
+        "php": "^7.1",
+        "psr/log": "^1.0"
+    },
     "require-dev": {
-        "php": "^7.0",
-        "psr/log": "^1.0",
         "doctrine/orm": "^2.5",
-        "predis/predis": "^1.1",
         "doctrine/phpcr-odm": "^1.4",
         "jackalope/jackalope-doctrine-dbal": "^1.2",
```
Comment (core23) [inline composer.json]: Do we need this? Couldn't find any usage of `Jackalope`

Comment (jordisala1991) [reply composer.json]: https://github.com/jackalope/jackalope-doctrine-dbal

this is in order to use phpcr, as you can see on suggest of phpcr-odm:

https://github.com/doctrine/phpcr-odm/blob/master/composer.json

File: composer.json (hunk 1 above)
Comment (jordisala1991) [reply composer.json]: Same reason it was removed here: https://github.com/sonata-project/SonataAdminBundle/pull/4295

**Comment by jordisala1991** [comment]
Will re-do my commits in a better way asap

File: composer.json (hunk 1 above)
Comment (core23) [reply composer.json]: You're right

**Comment by jordisala1991** [comment]
Now the commits should be better, please review @sonata-project/contributors

**Comment by greg0ire** [comment]
The build fails though

**Comment by jordisala1991** [comment]
You're right, we can remove autoload-dev only if autoload already covers those directories. On this repository we are using src / tests structure which is different from the Sonata Bundles.

Removed that commit, not it should pass.

Also added some phpunit.xml.dist modifications just to see if it changes something on the coverage report

**Comment by greg0ire** [comment]
> we can remove autoload-dev only if autoload already covers those directories

No, the right rule is remove it if you don't need to autoload any test class. Here, we extend a tests class `BaseTest` or whatever, and we need to autoload it. If there was no base test class, no autoload would work just fine.

**Comment by OskarStark** [comment]
Thank you @jordisala1991 👍
